packages/core/src/Account.ts

```diff
@@ -198,6 +209,57 @@ export class Account extends TypedEventEmitter<Events> {
     return storeEngine.updateOrCreate(AUTH_TABLE_NAME, AUTH_COOKIE_KEY, entity);
   }
 
+  private getCoreCryptoClient(): CoreCrypto | undefined {
+    if (this.cryptoClientDef) {
+      const [, cryptoClient] = this.cryptoClientDef;
+      const isCoreCrypto = (client: CoreCrypto | Cryptobox): client is CoreCrypto => client instanceof CoreCrypto;
+      const client = cryptoClient.getNativeClient();
+
+      if (isCoreCrypto(client)) {
+        return client;
+      }
+    }
+    return undefined;
+  }
+
+  public async enrollE2EI(displayName: string, handle: string, discoveryUrl: string): Promise<boolean> {
+    const client = this.getCoreCryptoClient();
+    const context = this.apiClient.context;
+    const domain = context?.domain ?? '';
+
+    if (!client) {
+      this.logger.info('Not a core crypto client, skipping E2EI enrollment', this.isMlsEnabled());
+      return false;
+    }
+
+    const user: User = {
+      displayName,
+      handle,
+      domain,
+      id: this.userId,
+    };
+    try {
+      const instance = await E2eIdentityService.getInstance({
+        apiClient: this.apiClient,
+        coreCryptClient: client,
+        user,
+        clientId: this.clientId,
+        discoveryUrl,
+        keyPackagesAmount: this.nbPrekeys,
+      });
+      const data = await instance.getNewCertificate();
+      if (data !== undefined) {
+        await this.service?.mls?.deleteMLSKeyPackages(data.keyPackageRefsToRemove, this.clientId);
+        await this.service?.mls?.uploadMLSKeyPackages(data.newKeyPackages, this.clientId);
+        return true;
+      }
+      return false;
+    } catch (error) {
+      this.logger.error('E2EI - Failed to enroll', error);
+      throw error;
+    }
```

Comment:
This piece of code should live in the `MLSService`. 
on the `Account` class you can just check that `this.service.mls` is not undefined, then you have the guarantee that you are in an MLS capable env. 

The reference to `cryptoClientDef` should be removed from the `Account` class